### DISCUSSION TITLE
v3.0.0

### DISCUSSION THREAD

**Discussion by gnikit** [krvajal/vscode-fortran-support#524] [Announcements]
### Added

- Added `Whats New` page to be displayed upon update of the extension.
- Added `ifx` support Intel's LLVM based compiler `ifx`
- Added ability to rescan for linting include files.
- Added GitHub Actions support for pre-Release builds ([#459](https://github.com/fortran-lang/vscode-fortran-support/issues/459))
- Added unittest for `fortls` spawning and integration, checks for initialization values ([#422](https://github.com/fortran-lang/vscode-fortran-support/issues/422))
- Added warning notifications for extensions that interfere with Modern Fortran ([#458](https://github.com/fortran-lang/vscode-fortran-support/issues/458))
- Added single file and multiple workspace folder support for the Language Server ([#446](https://github.com/fortran-lang/vscode-fortran-support/issues/446))
- Added file synchronization with VS Code settings and `.fortls` against the Language Server
- Added unittests for the formatting providers ([#423](https://github.com/fortran-lang/vscode-fortran-support/issues/423))
- Added GitHub Actions environment to dependabot
- Adds support for Intel (ifort) and LLVM (flang) compilers ([#291](https://github.com/fortran-lang/vscode-fortran-support/issues/291))
- Adds native support for the fortran-language-server (`fortls`) making unnecessary the usage of Fortran Intellisense extension ([#290](https://github.com/fortran-lang/vscode-fortran-support/issues/290))
- Added command for restarting the Language Server
- Added more options for configuring the `fortls` settings through the UI

### Changed

- Prettified the format of the settings UI and added `fortls` options
- Changed images from SVG to PNG because `vsce` does not support SVG ([#510](https://github.com/fortran-lang/vscode-fortran-support/pull/510))
- Changed need for matching begin-end scope names, in the following constructs:
  Functions, Modules, Programs, Module Procedures, Subroutines, Submodules.
  For a more detailed explanation as to why see the issue ([#278](https://github.com/fortran-lang/vscode-fortran-support/issues/278))
- Rewrote README to include links to fortran-lang and other projects ([#485](https://github.com/fortran-lang/vscode-fortran-support/issues/485)) ([#501](https://github.com/fortran-lang/vscode-fortran-support/issues/501))
- Changed `linter.compilerPath` to use the full path to the compiler instead of the root ([#500](https://github.com/fortran-lang/vscode-fortran-support/issues/500))
- Changed all instances of the publisher to `fortran-lang` ([#450](https://github.com/fortran-lang/vscode-fortran-support/issues/450))
- Updated grammar unittests to include scope injections
- Merged Language Server's log channel to Modern Fortran's log channel
- Merged all Fortran intrinsics into a single `json` file ([#424](https://github.com/fortran-lang/vscode-fortran-support/issues/424))
- Updates `README` text and animations, changes `SECURITY` and updates `package.json`
- Changes the interface of the extension to accommodate for the newest features ([#292](https://github.com/fortran-lang/vscode-fortran-support/issues/292))
- Changes main parts of the extension to being asynchronous ([#285](https://github.com/fortran-lang/vscode-fortran-support/issues/285))
- Changes Language Server prompt from `fortran-language-server` to `fortls`
- Updates VS Code engine to handle `vsce --pre-release`

### Removed

- Made redundant the use of FORTRAN Intellisense extension ([#290](https://github.com/fortran-lang/vscode-fortran-support/issues/290))
- Removed setting `includePaths` in favour of `linter.includePaths`
- Removed setting `gfortranExecutable` in favour of `linter.compilerPath`
- Removed setting `linterEnabled` in favour of `linter.compiler == Disabled`
- Removed setting `linterExtraArgs` in favour of `linter.extraArgs`
- Removed setting `linterModOutput` in favour of `linter.modOutput`
- Removed setting `ProvideSymbols` in favour of `provide.symbols`
- Removed setting `symbols`
- Removed setting `provideHover` in favour of `provide.hover`
- Removed setting `provideCompletion` in favour of `provide.autocomplete`

### Fixed

- Fixed formatter output mixes stdout and stderr ([#517](https://github.com/fortran-lang/vscode-fortran-support/issues/517))
- Fixed `error stop variable` syntax highlighting ([#486](https://github.com/fortran-lang/vscode-fortran-support/issues/486))
- Fixed issue with linter cache containing outdated folders ([#464](https://github.com/fortran-lang/vscode-fortran-support/issues/464))
- Fixed slow performance of very long lines by using a different solution for ([#207](https://github.com/fortran-lang/vscode-fortran-support/issues/207)) ([#309](https://github.com/fortran-lang/vscode-fortran-support/issues/309))
- Fixed hovering over user defined types while debugging ([#426](https://github.com/fortran-lang/vscode-fortran-support/issues/426))
- Fixes linting regex to capture a wider spectrum of errors ([#295](https://github.com/fortran-lang/vscode-fortran-support/issues/295))
- Fixes linter activation from `Disabled` to some compiler `X` without having to restart the extension ([#296](https://github.com/fortran-lang/vscode-fortran-support/issues/296))
- Fixes nopass pointer erroneous syntax highlighting ([#318](https://github.com/fortran-lang/vscode-fortran-support/issues/318))
- Fixes `%` accessor highlighting for type-bound subroutines ([#325](https://github.com/fortran-lang/vscode-fortran-support/issues/325))
- Fixes `fortls` not spawning when `ignoreWarning` was set to true ([#365](https://github.com/fortran-lang/vscode-fortran-support/issues/365))
- Fixes formatting on Windows (needed .exe extension) ([#354](https://github.com/fortran-lang/vscode-fortran-support/issues/354))
- Fixes `onSave` formatting errors ([#364](https://github.com/fortran-lang/vscode-fortran-support/issues/364))

### Security

- Updated node dependencies to the latest version
- Update GitHub Actions workflows to the latest version

**Full Changelog**: https://github.com/fortran-lang/vscode-fortran-support/compare/v2.6.2...v3.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/fortran-lang/vscode-fortran-support/releases/tag/v3.0.0'>v3.0.0</a>.</em>

**Comment by gbogopolsky** [comment]
Thank you for all your great work in reworking the Fortran ecosystem for VSCode!
A single extension, automatic installation of fortls, the new features, everything.
Congrats!

**Reply by gnikit** [reply]
Although, it would appear that we are still waiting to go online on the VS Code Marketplace

**Reply by gbogopolsky** [reply]
Ahhh I was wondering what was the problem, and why the version number was the pre-release one ^^.
Good catch

**Reply by gnikit** [reply]
Yeap, it was a mixup with Microsoft when we changed the publisher ID. I had to release v3.1.0 to have the chronologically newest release to also appear as the latest release in the VS Store. 

Technically speaking you could still install v3.0.0 from the Install Another Version menu, but it is easier to just issue a higher version number release.

**Reply by gbogopolsky** [reply]
Yup, I saw that I could do that, but I did not know if it would have frozen the version. 
It automatically updated to 3.1.0, so I agree with you, this was a good solution.
